src/foam/nanos/auth/CachedAuthServiceProxy.js

```diff
@@ -13,19 +13,33 @@ foam.CLASS({
     invalidated whenever the imported user's group changes.
   `,
   imports: [
+    'capabilityDAO',
     'user',
+    'userCapabilityJunctionDAO'
   ],
   properties: [
     {
       class: 'Map',
-      name: 'cache',
+      name: 'cache'
     },
   ],
   methods: [
     function init() {
       this.onDetach(this.user$.dot('group').sub(function() {
         this.cache = {};
       }.bind(this)));
+      this.onDetach(this.userCapabilityJunctionDAO.on.sub(
+        (sub, _on, event, ucj) => {
+          if ( event !== 'put' && event !== 'remove' ) return;
+          this.capabilityDAO.find(ucj.targetId).then(cap => {
+            var permissions = [
```

Comment:
Why copy values to just walk over them. Why not just:
cap.permissionsGranted.forEach(p => delete this.cache[p]);
cap.permissionsIntercepted.forEach(p => delete this.cache[p]);

Reply:
changes pushed https://github.com/Nauna/foam2/pull/82